<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="jeprob" />

<meta name="date" content="2021-01-07" />

<title>Analysis with Seurat</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">project-scrnaseq-integration-comparison</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="analysiswithseurat.html">Seurat</a>
</li>
<li>
  <a href="analysiswithconos.html">Conos</a>
</li>
<li>
  <a href="analysiswithharmony.html">Harmony</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jeprob/project-scrnaseq-integration-comparison">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Analysis with Seurat</h1>
<h4 class="author">jeprob</h4>
<h4 class="date">2021-01-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>project-scrnaseq-integration-comparison/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210107code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210107)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210107code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210107)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjeprobprojectscrnaseqintegrationcomparisontreec0766037f548e1e172a3ca219d09ba221510d827targetblankc076603a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jeprob/project-scrnaseq-integration-comparison/tree/c0766037f548e1e172a3ca219d09ba221510d827" target="_blank">c076603</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjeprobprojectscrnaseqintegrationcomparisontreec0766037f548e1e172a3ca219d09ba221510d827targetblankc076603a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jeprob/project-scrnaseq-integration-comparison/tree/c0766037f548e1e172a3ca219d09ba221510d827" target="_blank">c076603</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
working directory clean
</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/analysiswithseurat.Rmd</code>) and HTML (<code>docs/analysiswithseurat.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jeprob/project-scrnaseq-integration-comparison/blob/751203d6d80b577eeada709fff7d744afbdd7d52/analysis/analysiswithseurat.Rmd" target="_blank">751203d</a>
</td>
<td>
jeprob
</td>
<td>
2021-01-07
</td>
<td>
.gitignore + Rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="seurat-integration" class="section level1">
<h1>Seurat integration:</h1>
<p>Dataset preprocessing and integration was adapted from <a href="https://satijalab.org/seurat/v3.0/integration.html">this vignette</a> by the developers of Seurat. Furthermore this <a href="https://satijalab.org/seurat/v3.2/pbmc3k_tutorial.html">tutorial</a> for pbmcs was followed for cell clustering.</p>
<div id="load-packages" class="section level2">
<h2>Load packages</h2>
<pre class="r"><code>library(dplyr)
library(ggplot2)
library(Seurat)
library(cowplot)
library(SeuratDisk)
library(SeuratData)</code></pre>
</div>
<div id="read-in-the-data" class="section level2">
<h2>Read in the data</h2>
<pre class="r"><code>data_pbmc &lt;- readRDS(&quot;./data/wonglab_10x_XLP2_sce_after_scDblFinder.rds&quot;)
data_pbmc</code></pre>
<pre><code>class: SingleCellExperiment 
dim: 10081 46299 
metadata(2): Samples scDblFinder.stats
assays(2): counts logcounts
rownames(10081): ENSG00000237491.AL669831.5 ENSG00000230368.FAM41C ...
  ENSG00000198727.MT-CYB ENSG00000273748.AL592183.1
rowData names(3): ID Symbol Type
colnames(46299): het1.AAACCTGAGATGTTAG-1 het1.AAACCTGAGGTGGGTT-1 ...
  wt2.TTTGTCATCGCCGTGA-1 wt2.TTTGTCATCTTGACGA-1
colData names(28): Sample Barcode ... scDblFinder.originAmbiguous
  cluster
reducedDimNames(2): PCA UMAP
altExpNames(0):</code></pre>
<pre class="r"><code>head(counts(data_pbmc))</code></pre>
<pre><code>6 x 46299 sparse Matrix of class &quot;dgCMatrix&quot;
                                                                              
ENSG00000237491.AL669831.5 . . . . . . . . . . . . . . . . . . . . . . . . . .
ENSG00000230368.FAM41C     . . . . . . . . . . . . . . . . . . . . . . . . . .
ENSG00000188976.NOC2L      1 . 1 . . . . . . . . . . . 1 . 1 . . . . . . . . .
ENSG00000188290.HES4       . . . . . . 2 . . . . . . . . . . . . . . . . . . .
ENSG00000187608.ISG15      . 1 . . . . 6 . . . 1 . . . . 1 . . . . . 1 . . . .
ENSG00000186891.TNFRSF18   . . . . . . . . . . . . . . . . . . . . . . . . . .
                                                 
ENSG00000237491.AL669831.5 . . . . . . . . ......
ENSG00000230368.FAM41C     . . . . . . . . ......
ENSG00000188976.NOC2L      1 2 . . . . 1 . ......
ENSG00000188290.HES4       . . . . . . . . ......
ENSG00000187608.ISG15      . . . . . . . . ......
ENSG00000186891.TNFRSF18   . . . . . . . . ......

 .....suppressing 46265 columns in show(); maybe adjust &#39;options(max.print= *, width = *)&#39;
 ..............................</code></pre>
</div>
<div id="dataset-preprocessing" class="section level2">
<h2>Dataset preprocessing</h2>
<pre class="r"><code>#Set up Seurat object
pbmc &lt;- as.Seurat(x = data_pbmc)
# pbmc &lt;- Seurat::NormalizeData(pbmc, verbose=FALSE)

#Split the seurat object into individual datasets divided by Sample (in metadata)
pbmc.list &lt;- SplitObject(pbmc, split.by = &#39;Sample&#39;) 

#Identify variable features with variance stabilizing transformation (feature selection) for each dataset
for (i in 1:length(pbmc.list)) {
    pbmc.list[[i]] &lt;- NormalizeData(pbmc.list[[i]]) %&gt;%  FindVariableFeatures(selection.method = &quot;vst&quot;, nfeatures = 2000, verbose = FALSE)  %&gt;% ScaleData()
}

#just for visualization purposes: 
#Identify the 1000 most highly variable genes from 1st dataset and plot variable features with and without labels
top1000 &lt;- head(VariableFeatures(pbmc.list[[1]]), 1000)
plot1 &lt;- VariableFeaturePlot(pbmc.list[[1]])
plot1 </code></pre>
<p><img src="figure/analysiswithseurat.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /> One can see in the plot that the 2000 most variable features chosen by the VariableFeatures function are actually the ones displaying the highest standardized variance.</p>
</div>
<div id="integration-of-the-six-datasets" class="section level2">
<h2>Integration of the six datasets</h2>
<pre class="r"><code>#Find anchors between individual datasets to construct a reference
pbmc.anchors &lt;- FindIntegrationAnchors(pbmc.list, dims=1:30)

#Integrate data (batch corrected expression matrix)
pbmc.integrated &lt;- IntegrateData(anchorset=pbmc.anchors, dims=1:30)

#save and reload data
# SaveH5Seurat(pbmc.integrated, &#39;integrated_seurat&#39;, overwrite = FALSE, verbose = TRUE)
# pbmc.integrated &lt;- LoadH5Seurat(&quot;../integrated_seurat.h5Seurat&quot;)</code></pre>
<p>In the last iteration of the FindIntegrationAnchors function 17359 anchors were found and 7623 anchors retained, to use for integration.</p>
</div>
<div id="downstream-analysis" class="section level2">
<h2>Downstream analysis</h2>
<p>For comparison we perform the same UMAP and Nearest Neighbor analyses with the embeddings from the Seurat batch corrected gene expression matrix (integrated datasets) and the uncorrected initial PC embeddings (original datasets).</p>
<p>We start with the analysis of the integrated dataset: <img src="figure/analysiswithseurat.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /> We an see that the data is well integrated, and cells do not cluster by dataset in the UMAP space.</p>
<p>Now we aim to identify different clusters from the integrated dataset in the UMAP embedding.</p>
<pre class="r"><code>#find clusters
pbmc.integrated &lt;- FindNeighbors(pbmc.integrated, dims = 1:10)
pbmc.integrated &lt;- FindClusters(pbmc.integrated, resolution = 0.8)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 46299
Number of edges: 1506510

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8992
Number of communities: 20
Elapsed time: 21 seconds</code></pre>
<pre class="r"><code>head(Idents(pbmc.integrated), 5)</code></pre>
<pre><code>het1.AAACCTGAGATGTTAG-1 het1.AAACCTGAGGTGGGTT-1 het1.AAACCTGAGTGGGTTG-1 
                      1                      13                       3 
het1.AAACCTGAGTGTCTCA-1 het1.AAACCTGCAAGTTGTC-1 
                      0                       0 
Levels: 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19</code></pre>
<pre class="r"><code>pbmc.integrated$newcluster &lt;- as.factor(Idents(pbmc.integrated))

#plot by cluster
p2 &lt;- DimPlot(pbmc.integrated, reduction = &quot;umap&quot;, group.by = &quot;newcluster&quot;, label = TRUE, repel = TRUE)
p2</code></pre>
<p><img src="figure/analysiswithseurat.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /> We find 20 clusters identified from the Seurat integrated embeddings.</p>
<p>Now we run the same analysis for the original dataset without integration. The only thing that changes is that one first has to find variable features on the whole unprocessed dataset, which had already been done for the integrated dataset.</p>
<pre class="r"><code>#standard workflow for visualization
pbmc &lt;- FindVariableFeatures(pbmc, nfeatues=2000)
pbmc &lt;- ScaleData(pbmc, verbose = FALSE)
pbmc &lt;- RunPCA(pbmc, npcs = 30, verbose = FALSE)
pbmc &lt;- RunUMAP(pbmc, reduction = &quot;pca&quot;, dims = 1:30)
p1_bef &lt;- DimPlot(pbmc, reduction = &quot;umap&quot;, group.by = &quot;Sample&quot;)
p1_bef</code></pre>
<p><img src="figure/analysiswithseurat.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /> We see, that the original dataset is not well integrated and cells cluster by patient (het1 to wt2) in the UMAP plot above.</p>
<p>As previously done, we have a look at the clusters that can be identified from the original unprocessed embeddings in the UMAP space.</p>
<pre class="r"><code>#find clusters
pbmc &lt;- FindNeighbors(pbmc, dims = 1:10)
pbmc &lt;- FindClusters(pbmc, resolution = 0.8)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 46299
Number of edges: 1407226

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8901
Number of communities: 19
Elapsed time: 22 seconds</code></pre>
<pre class="r"><code>head(Idents(pbmc), 5)</code></pre>
<pre><code>het1.AAACCTGAGATGTTAG-1 het1.AAACCTGAGGTGGGTT-1 het1.AAACCTGAGTGGGTTG-1 
                      2                       7                       3 
het1.AAACCTGAGTGTCTCA-1 het1.AAACCTGCAAGTTGTC-1 
                      0                       0 
Levels: 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18</code></pre>
<pre class="r"><code>pbmc$newcluster &lt;- as.factor(Idents(pbmc))

#plot by cluster
p2_bef &lt;- DimPlot(pbmc, reduction = &quot;umap&quot;, group.by = &quot;newcluster&quot;, label = TRUE, repel = TRUE)
p2_bef</code></pre>
<p><img src="figure/analysiswithseurat.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /> We note that in comparison to the 20 identified clusters from the Seurat embedding, only 19 clusters could be recognized from the original UMAP embeddings.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=de_CH.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=de_CH.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=de_CH.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=de_CH.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0
 [3] Biobase_2.50.0              GenomicRanges_1.42.0       
 [5] GenomeInfoDb_1.26.2         IRanges_2.24.1             
 [7] S4Vectors_0.28.1            BiocGenerics_0.36.0        
 [9] MatrixGenerics_1.2.0        matrixStats_0.57.0         
[11] SeuratData_0.2.1            SeuratDisk_0.0.0.9013      
[13] cowplot_1.1.1               Seurat_3.2.3               
[15] ggplot2_3.3.3               dplyr_1.0.2                
[17] workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] plyr_1.8.6             igraph_1.2.6           lazyeval_0.2.2        
  [4] splines_4.0.3          listenv_0.8.0          scattermore_0.7       
  [7] digest_0.6.27          htmltools_0.5.0        fansi_0.4.1           
 [10] magrittr_2.0.1         tensor_1.5             cluster_2.1.0         
 [13] ROCR_1.0-11            globals_0.14.0         colorspace_2.0-0      
 [16] rappdirs_0.3.1         ggrepel_0.9.0          xfun_0.19             
 [19] crayon_1.3.4           RCurl_1.98-1.2         jsonlite_1.7.2        
 [22] spatstat_1.64-1        spatstat.data_1.7-0    survival_3.2-7        
 [25] zoo_1.8-8              glue_1.4.2             polyclip_1.10-0       
 [28] gtable_0.3.0           zlibbioc_1.36.0        XVector_0.30.0        
 [31] leiden_0.3.6           DelayedArray_0.16.0    future.apply_1.7.0    
 [34] abind_1.4-5            scales_1.1.1           miniUI_0.1.1.1        
 [37] Rcpp_1.0.5             viridisLite_0.3.0      xtable_1.8-4          
 [40] reticulate_1.18        bit_4.0.4              rsvd_1.0.3            
 [43] htmlwidgets_1.5.3      httr_1.4.2             RColorBrewer_1.1-2    
 [46] ellipsis_0.3.1         ica_1.0-2              pkgconfig_2.0.3       
 [49] farver_2.0.3           uwot_0.1.10            deldir_0.2-3          
 [52] tidyselect_1.1.0       labeling_0.4.2         rlang_0.4.10          
 [55] reshape2_1.4.4         later_1.1.0.1          munsell_0.5.0         
 [58] tools_4.0.3            cli_2.2.0              generics_0.1.0        
 [61] ggridges_0.5.2         evaluate_0.14          stringr_1.4.0         
 [64] fastmap_1.0.1          yaml_2.2.1             goftest_1.2-2         
 [67] knitr_1.30             bit64_4.0.5            fs_1.5.0              
 [70] fitdistrplus_1.1-3     purrr_0.3.4            RANN_2.6.1            
 [73] pbapply_1.4-3          future_1.21.0          nlme_3.1-151          
 [76] whisker_0.4            mime_0.9               hdf5r_1.3.3           
 [79] compiler_4.0.3         rstudioapi_0.13        plotly_4.9.2.2        
 [82] png_0.1-7              spatstat.utils_1.17-0  tibble_3.0.4          
 [85] stringi_1.5.3          RSpectra_0.16-0        lattice_0.20-41       
 [88] Matrix_1.3-0           vctrs_0.3.6            pillar_1.4.7          
 [91] lifecycle_0.2.0        lmtest_0.9-38          RcppAnnoy_0.0.18      
 [94] data.table_1.13.6      bitops_1.0-6           irlba_2.3.3           
 [97] httpuv_1.5.4           patchwork_1.1.1        R6_2.5.0              
[100] promises_1.1.1         KernSmooth_2.23-18     gridExtra_2.3         
[103] parallelly_1.23.0      codetools_0.2-18       MASS_7.3-53           
[106] assertthat_0.2.1       rprojroot_2.0.2        withr_2.3.0           
[109] sctransform_0.3.2      GenomeInfoDbData_1.2.4 mgcv_1.8-33           
[112] grid_4.0.3             rpart_4.1-15           tidyr_1.1.2           
[115] rmarkdown_2.6          Rtsne_0.15             git2r_0.27.1          
[118] shiny_1.5.0           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
