<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="jeprob" />

<meta name="date" content="2021-01-07" />

<title>Analysis with Conos</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">project-scrnaseq-integration-comparison</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jeprob/project-scrnaseq-integration-comparison">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Analysis with Conos</h1>
<h4 class="author">jeprob</h4>
<h4 class="date">2021-01-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>project-scrnaseq-integration-comparison/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210107code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210107)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210107code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210107)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjeprobprojectscrnaseqintegrationcomparisontreed5ff520884c61bc36ec504710f1c74179a6bb1aetargetblankd5ff520a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jeprob/project-scrnaseq-integration-comparison/tree/d5ff520884c61bc36ec504710f1c74179a6bb1ae" target="_blank">d5ff520</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjeprobprojectscrnaseqintegrationcomparisontreed5ff520884c61bc36ec504710f1c74179a6bb1aetargetblankd5ff520a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jeprob/project-scrnaseq-integration-comparison/tree/d5ff520884c61bc36ec504710f1c74179a6bb1ae" target="_blank">d5ff520</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/analysiswithharmony.Rmd
    Untracked:  analysis/analysiswithseurat.Rmd

Unstaged changes:
    Modified:   .gitignore

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/analysiswithconos.Rmd</code>) and HTML (<code>docs/analysiswithconos.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jeprob/project-scrnaseq-integration-comparison/blob/d5ff520884c61bc36ec504710f1c74179a6bb1ae/analysis/analysiswithconos.Rmd" target="_blank">d5ff520</a>
</td>
<td>
jeprob
</td>
<td>
2021-01-07
</td>
<td>
wflow_publish(“analysis/analysiswithconos.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="conos-integration" class="section level1">
<h1>Conos integration:</h1>
<p>Dataset preprocessing and integration was adapted from <a href="http://htmlpreview.github.io/?https://github.com/satijalab/seurat-wrappers/blob/master/docs/conos.html">this vignette</a> on the Seurat github. The Conos integration followed <a href="https://github.com/kharchenkolab/conos/blob/master/vignettes/walkthrough.md">this vignette</a> by the developers of Conos, subsection: ‘Systematic comparative analysis of human PBMC’.</p>
<div id="load-packages" class="section level2">
<h2>Load packages</h2>
<pre class="r"><code>library(conos)
library(dplyr)
library(pagoda2)
library(Seurat) </code></pre>
</div>
<div id="read-in-the-data-and-set-up-seurat-object" class="section level2">
<h2>Read in the data and set up Seurat object</h2>
<pre class="r"><code>data_pbmc &lt;- readRDS(&quot;./data/wonglab_10x_XLP2_sce_after_scDblFinder.rds&quot;)
data_pbmc</code></pre>
<pre><code>class: SingleCellExperiment 
dim: 10081 46299 
metadata(2): Samples scDblFinder.stats
assays(2): counts logcounts
rownames(10081): ENSG00000237491.AL669831.5 ENSG00000230368.FAM41C ...
  ENSG00000198727.MT-CYB ENSG00000273748.AL592183.1
rowData names(3): ID Symbol Type
colnames(46299): het1.AAACCTGAGATGTTAG-1 het1.AAACCTGAGGTGGGTT-1 ...
  wt2.TTTGTCATCGCCGTGA-1 wt2.TTTGTCATCTTGACGA-1
colData names(28): Sample Barcode ... scDblFinder.originAmbiguous
  cluster
reducedDimNames(2): PCA UMAP
altExpNames(0):</code></pre>
<pre class="r"><code>head(counts(data_pbmc))</code></pre>
<pre><code>6 x 46299 sparse Matrix of class &quot;dgCMatrix&quot;
                                                                              
ENSG00000237491.AL669831.5 . . . . . . . . . . . . . . . . . . . . . . . . . .
ENSG00000230368.FAM41C     . . . . . . . . . . . . . . . . . . . . . . . . . .
ENSG00000188976.NOC2L      1 . 1 . . . . . . . . . . . 1 . 1 . . . . . . . . .
ENSG00000188290.HES4       . . . . . . 2 . . . . . . . . . . . . . . . . . . .
ENSG00000187608.ISG15      . 1 . . . . 6 . . . 1 . . . . 1 . . . . . 1 . . . .
ENSG00000186891.TNFRSF18   . . . . . . . . . . . . . . . . . . . . . . . . . .
                                                 
ENSG00000237491.AL669831.5 . . . . . . . . ......
ENSG00000230368.FAM41C     . . . . . . . . ......
ENSG00000188976.NOC2L      1 2 . . . . 1 . ......
ENSG00000188290.HES4       . . . . . . . . ......
ENSG00000187608.ISG15      . . . . . . . . ......
ENSG00000186891.TNFRSF18   . . . . . . . . ......

 .....suppressing 46265 columns in show(); maybe adjust &#39;options(max.print= *, width = *)&#39;
 ..............................</code></pre>
</div>
<div id="set-up-a-seurat-object-and-preprocess-with-seurat" class="section level2">
<h2>Set up a Seurat object and preprocess with Seurat</h2>
<pre class="r"><code>#set up seurat object
pbmc &lt;- as.Seurat(x=data_pbmc)
pbmc</code></pre>
<pre><code>An object of class Seurat 
10081 features across 46299 samples within 1 assay 
Active assay: RNA (10081 features, 0 variable features)
 2 dimensional reductions calculated: PCA, UMAP</code></pre>
<pre class="r"><code>#split into list of individual datasets
pbmc.panel &lt;- SplitObject(pbmc, split.by = &#39;Sample&#39;)
str(pbmc.panel,1)</code></pre>
<pre><code>List of 6
 $ het1:Formal class &#39;Seurat&#39; [package &quot;Seurat&quot;] with 13 slots
 $ ko1 :Formal class &#39;Seurat&#39; [package &quot;Seurat&quot;] with 13 slots
 $ m2  :Formal class &#39;Seurat&#39; [package &quot;Seurat&quot;] with 13 slots
 $ p2  :Formal class &#39;Seurat&#39; [package &quot;Seurat&quot;] with 13 slots
 $ wt1 :Formal class &#39;Seurat&#39; [package &quot;Seurat&quot;] with 13 slots
 $ wt2 :Formal class &#39;Seurat&#39; [package &quot;Seurat&quot;] with 13 slots</code></pre>
</div>
<div id="preprocessing-with-seurat" class="section level2">
<h2>Preprocessing with Seurat</h2>
<p>Normal Seurat preprocessing steps are NormalizeDate, FindVariableFeatures, ScaleData and run PCA and TSNE for each dataset.</p>
<pre class="r"><code>for (i in 1:length(pbmc.panel)){
  pbmc.panel[[i]] &lt;- NormalizeData(pbmc.panel[[i]]) %&gt;% FindVariableFeatures() %&gt;% ScaleData() %&gt;% RunPCA(verbose=FALSE) %&gt;% RunTSNE(verbose=FALSE)
}</code></pre>
</div>
<div id="integrating-dataset-with-conos" class="section level2">
<h2>Integrating dataset with Conos</h2>
<pre class="r"><code># construct Conos object for the panel of samples
pbmc.con &lt;- Conos$new(pbmc.panel)
pbmc.con #have a look at the Conos object</code></pre>
<pre><code>&lt;Conos&gt;
  Public:
    addSamples: function (x, replace = FALSE, verbose = FALSE) 
    buildGraph: function (k = 15, k.self = 10, k.self.weight = 0.1, alignment.strength = NULL, 
    clone: function (deep = FALSE) 
    clusters: list
    correctGenes: function (genes = NULL, n.od.genes = 500, fading = 10, fading.const = 0.5, 
    embedding: NULL
    embedGraph: function (method = &quot;largeVis&quot;, M = 1, gamma = 1, alpha = 0.1, 
    expression.adj: list
    findCommunities: function (method = leiden.community, min.group.size = 0, name = NULL, 
    getClusterCountMatrices: function (clustering = NULL, groups = NULL, common.genes = TRUE, 
    getDatasetPerCell: function () 
    getDifferentialGenes: function (clustering = NULL, groups = NULL, z.threshold = 3, 
    getJointCountMatrix: function (raw = FALSE) 
    graph: NULL
    initialize: function (x, ..., n.cores = parallel::detectCores(logical = FALSE), 
    misc: list
    n.cores: 8
    override.conos.plot.theme: FALSE
    pairs: list
    plotClusterStability: function (clustering = NULL, what = &quot;all&quot;) 
    plotGraph: function (color.by = &quot;cluster&quot;, clustering = NULL, groups = NULL, 
    plotPanel: function (clustering = NULL, groups = NULL, colors = NULL, gene = NULL, 
    propagateLabels: function (labels, method = &quot;diffusion&quot;, ...) 
    samples: list
  Private:
    adjustTheme: function (theme) 
    updatePairs: function (space = &quot;PCA&quot;, data.type = &quot;counts&quot;, ncomps = 50, n.odgenes = 1000, </code></pre>
<pre class="r"><code>str(pbmc.con$samples,1)</code></pre>
<pre><code>List of 6
 $ het1:Formal class &#39;Seurat&#39; [package &quot;Seurat&quot;] with 13 slots
 $ ko1 :Formal class &#39;Seurat&#39; [package &quot;Seurat&quot;] with 13 slots
 $ m2  :Formal class &#39;Seurat&#39; [package &quot;Seurat&quot;] with 13 slots
 $ p2  :Formal class &#39;Seurat&#39; [package &quot;Seurat&quot;] with 13 slots
 $ wt1 :Formal class &#39;Seurat&#39; [package &quot;Seurat&quot;] with 13 slots
 $ wt2 :Formal class &#39;Seurat&#39; [package &quot;Seurat&quot;] with 13 slots</code></pre>
<p>We now build the joint graph:</p>
<pre class="r"><code>pbmc.con$buildGraph(k=15, k.self=5, space=&#39;PCA&#39;, ncomps=30, n.odgenes=2000, matching.method=&#39;mNN&#39;, metric=&#39;angular&#39;, score.component.variance=TRUE, verbose=TRUE)</code></pre>
<pre><code>.</code></pre>
<p>Next, we can visualize the estimated amount of variance explained by successive PCs:</p>
<pre class="r"><code>plotComponentVariance(pbmc.con, space=&#39;PCA&#39;)</code></pre>
<p><img src="figure/analysiswithconos.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /> We can see that the first PC already explaines a large fraction of the variance. The second and the third PC also explain a much higher fraction of variance than the remaining ones.</p>
<div id="downstream-analysis" class="section level3">
<h3>Downstream analysis</h3>
<p>We can now use the graph to find and visualize global clusters. Here, we use the Leiden community detection.</p>
<pre class="r"><code>pbmc.con$findCommunities(method=leiden.community, resolution=0.8)</code></pre>
<p>As a next step, the joint graph can be used to visualize the integration of datasets and the clustering.</p>
<pre class="r"><code>pbmc.con$plotGraph(color.by=&#39;sample&#39;, mark.groups=FALSE, alpha=0.1, show.legend=TRUE)</code></pre>
<pre><code>Estimating embeddings.</code></pre>
<p><img src="figure/analysiswithconos.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /> One can observe that the data is integrated in a joint TSNE space and cells do not cluster by dataset.</p>
<pre class="r"><code>pbmc.con$plotGraph(alpha=0.5)</code></pre>
<p><img src="figure/analysiswithconos.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /> We find that there are 14 clusters identified in total.</p>
<p>Next, the joint clustering for each panel are plotted, where cell labels obtained from the joint clustering are added in the original TSNE embedding.</p>
<pre class="r"><code>pbmc.con$plotPanel(font.size=4)</code></pre>
<p><img src="figure/analysiswithconos.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /> We find different amounts of cell types in the individual datasets. To have a closer look at the composition of the individual datasets, we plot cluster barplots for the individual datasets.</p>
<pre class="r"><code>plotClusterBarplots(pbmc.con, legend.height = 0.1)</code></pre>
<p><img src="figure/analysiswithconos.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /> Many interesting things can be observed in this plot. It is noteable that most of the cells in ko1 belong to cluster 2. Furthermore, the distribution of cells to clusters differs from wt1 to wt2, judged by eye. Cluster 12 is not present in the dataset het1.</p>
<p>Further downstream analysis such as gene expression visualization or hierarchical community structure generation would be possible with Conos. For more information visit <a href="https://github.com/kharchenkolab/conos/blob/master/vignettes/walkthrough.md">this vignette</a>.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=de_CH.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=de_CH.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=de_CH.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=de_CH.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0
 [3] Biobase_2.50.0              GenomicRanges_1.42.0       
 [5] GenomeInfoDb_1.26.2         IRanges_2.24.1             
 [7] S4Vectors_0.28.1            BiocGenerics_0.36.0        
 [9] MatrixGenerics_1.2.0        matrixStats_0.57.0         
[11] Seurat_3.2.3                pagoda2_1.0.0              
[13] dplyr_1.0.2                 conos_1.3.1                
[15] igraph_1.2.6                Matrix_1.3-0               
[17] workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] N2R_0.1.1              circlize_0.4.11        plyr_1.8.6            
  [4] lazyeval_0.2.2         splines_4.0.3          entropy_1.2.1         
  [7] listenv_0.8.0          scattermore_0.7        ggplot2_3.3.3         
 [10] urltools_1.7.3         digest_0.6.27          htmltools_0.5.0       
 [13] viridis_0.5.1          magrittr_2.0.1         RMTstat_0.3           
 [16] tensor_1.5             cluster_2.1.0          ROCR_1.0-11           
 [19] ComplexHeatmap_2.7.4   globals_0.14.0         R.utils_2.10.1        
 [22] sccore_0.1.1           colorspace_2.0-0       ggrepel_0.9.0         
 [25] xfun_0.19              RCurl_1.98-1.2         crayon_1.3.4          
 [28] jsonlite_1.7.2         spatstat_1.64-1        spatstat.data_1.7-0   
 [31] brew_1.0-6             survival_3.2-7         zoo_1.8-8             
 [34] glue_1.4.2             polyclip_1.10-0        gtable_0.3.0          
 [37] zlibbioc_1.36.0        XVector_0.30.0         leiden_0.3.6          
 [40] DelayedArray_0.16.0    GetoptLong_1.0.5       Rook_1.1-1            
 [43] future.apply_1.7.0     shape_1.4.5            abind_1.4-5           
 [46] scales_1.1.1           miniUI_0.1.1.1         Rcpp_1.0.5            
 [49] viridisLite_0.3.0      xtable_1.8-4           clue_0.3-58           
 [52] reticulate_1.18        rsvd_1.0.3             htmlwidgets_1.5.3     
 [55] httr_1.4.2             RColorBrewer_1.1-2     ellipsis_0.3.1        
 [58] ica_1.0-2              farver_2.0.3           pkgconfig_2.0.3       
 [61] R.methodsS3_1.8.1      uwot_0.1.10            deldir_0.2-3          
 [64] labeling_0.4.2         tidyselect_1.1.0       rlang_0.4.10          
 [67] reshape2_1.4.4         later_1.1.0.1          munsell_0.5.0         
 [70] tools_4.0.3            generics_0.1.0         ggridges_0.5.2        
 [73] evaluate_0.14          stringr_1.4.0          fastmap_1.0.1         
 [76] yaml_2.2.1             goftest_1.2-2          knitr_1.30            
 [79] fs_1.5.0               fitdistrplus_1.1-3     purrr_0.3.4           
 [82] RANN_2.6.1             dendextend_1.14.0      pbapply_1.4-3         
 [85] future_1.21.0          nlme_3.1-151           whisker_0.4           
 [88] mime_0.9               R.oo_1.24.0            compiler_4.0.3        
 [91] rstudioapi_0.13        plotly_4.9.2.2         png_0.1-7             
 [94] spatstat.utils_1.17-0  tibble_3.0.4           stringi_1.5.3         
 [97] lattice_0.20-41        vctrs_0.3.6            pillar_1.4.7          
[100] lifecycle_0.2.0        triebeard_0.3.0        lmtest_0.9-38         
[103] GlobalOptions_0.1.2    RcppAnnoy_0.0.18       bitops_1.0-6          
[106] data.table_1.13.6      cowplot_1.1.1          irlba_2.3.3           
[109] httpuv_1.5.4           patchwork_1.1.1        R6_2.5.0              
[112] promises_1.1.1         KernSmooth_2.23-18     gridExtra_2.3         
[115] parallelly_1.23.0      codetools_0.2-18       MASS_7.3-53           
[118] rprojroot_2.0.2        rjson_0.2.20           withr_2.3.0           
[121] sctransform_0.3.2      GenomeInfoDbData_1.2.4 mgcv_1.8-33           
[124] grid_4.0.3             rpart_4.1-15           tidyr_1.1.2           
[127] rmarkdown_2.6          dendsort_0.3.3         Cairo_1.5-12.2        
[130] Rtsne_0.15             git2r_0.27.1           shiny_1.5.0           
[133] base64enc_0.1-3       </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
